pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2145,6 +2186,200 @@ func TestUpdateServiceWithConflictingNamespace(t *testing.T) {
 	}
 }
 
+func TestServiceRegistryCreateServiceMultipleNodePortsWithSameProtocol(t *testing.T) {
+	ctx := genericapirequest.NewDefaultContext()
+	storage, registry, server := NewTestREST(t, nil, singleStackIPv4)
+	defer server.Terminate(t)
+
+	const count = 3
+	// The `PortAllocator` seems to allocate the first port randomly,
+	// So we run the test multiple times to ensure the NodePort is correctly allocated.
```

Comment:
The code change seems good, but I am not sure I understand this random + retry.  Can you help me understand?

Reply:
The port allocator would allocate port randomly from the available ports, if we did not retry, the port allocator might happen to allocate the right port and the test got passed.